src/migrations/20240129221712-change-training-event-goal-source.js

```diff
@@ -8,28 +7,79 @@ module.exports = {
       const sessionSig = __filename;
       await prepMigration(queryInterface, transaction, sessionSig);
 
-      // Goals.
-      await dropAndRecreateEnum(
-        queryInterface,
-        transaction,
-        'enum_Goals_source',
-        'Goals',
-        'source',
-        GOAL_SOURCES,
-        'text',
-        false,
+      /* Goals */
+      // Add new enum value (Can't use transaction here).
+      await queryInterface.sequelize.query(`
+      DO $$ BEGIN
+        ALTER TYPE "enum_Goals_source" ADD VALUE 'Training event';
+      EXCEPTION
+        WHEN duplicate_object THEN null;
+      END $$;
+    `);
+
+      await queryInterface.sequelize.query(`
+    DO $$ BEGIN
+      ALTER TYPE "enum_Goals_source" ADD VALUE 'Training event follow-up';
+    EXCEPTION
+      WHEN duplicate_object THEN null;
+    END $$;
+  `);
+
+      // Update to new enum value.
+      await queryInterface.sequelize.query(
+        `UPDATE "Goals"
+            SET "source" = 'Training event'::"enum_Goals_source"
+          WHERE "source" = 'Training event follow-up'::"enum_Goals_source";`,
+        { transaction },
+      );
+
+      // Remove enum value.
+      await queryInterface.sequelize.query(
+        `DELETE FROM pg_enum
+            WHERE enumlabel = 'Training event follow-up'
+              AND enumtypid = (
+                SELECT oid
+                FROM pg_type
+                WHERE typname = 'enum_Goals_source'
+              );`,
+        { transaction },
+      );
```

Comment:
This is not safe, do not do it this way. enums must only be added to or dropped

Reply:
the pg_* tables are only for reading as they are generated. Do not edit them

Reply:
What is the risk?

Reply:
This is fully unsupported, and not guaranteed to function as you are using it. the pg_* tables are for reading only not writing.  Please do it the correct way

Reply:
With enums, the only operations permitted are create, append, or drop.

Reply:
`ALTER TYPE mood_enum RENAME VALUE 'ok' TO 'content';`

Reply:
Ok think I have the other way working doing one more test.